docker-compose.yml

```diff
@@ -21,6 +21,8 @@ services:
     networks:
       - api-network
       - redis-network
+    env_file:
+      - .env
```

Comment:
Not sure if this is a miss or if I'm missing something? Couldn't get `docker-compose up` to work without it. Or perhaps the REDIS_HOST etc. should just be set under the env section?

Reply:
Hmm, it's been working fine for me without setting it explicitly, and it looks like [that's expected](https://docs.docker.com/compose/environment-variables/#substitute-environment-variables-in-compose-files)

> You can set default values for environment variables using a [.env file](https://docs.docker.com/compose/env-file/), which Compose automatically looks for in project directory (parent folder of your Compose file). Values set in the shell environment override those set in the .env file.

I don't think referencing it explicitly is a problem?

Reply:
It was trying to connect to Redis on 127.0.0.1, which is the default when no env. var. is set. However, mystery is solved: I had installed docker-compose from the Fedora repo but that was some old legacy version. Installed the new plugin now (so that I can do `docker compose up`) and now it works as expected without specifying the env_file. That being said, part of what confused me was that the `pgsql` container had the `env_file` listed as well. Should perhaps be cleaned up as well :thinking:

Reply:
Jokes on me, that was too early of a celebration. It did not solve it. I tried both the plugin and the standalone, v2.11.2. Still not loading the env file automatically unless it's specified in the compose file

Reply:
Okay well lets leave it as is for now and see if it solves it self. 

Now that I'm thinking about it I think I actually encountered the same issue within the GitHub Action that it wasn't finding the env file in the test docker-compose. So seems safest to have it mentioned.

Reply:
For posterity: Found out the reason why this needs to be specified: https://stackoverflow.com/a/58048994/998919
Essentially, no env vars will be set inside the container if you don't specify the .env file. The reason why it might be working for you without setting the env vars might be because you're on a different OS that works a bit differently perhaps? Do you happen to be running macOS? I think only linux 100% "correctly" runs docker, both macOS and Windows fudges some things with the networking. Either way, if we expect the env vars to be used inside the containers, let's make sure to specify the "env_file".

Reply:
Clueless then. Is your api really able to connect to the db/Redis on 127.0.0.1?